sawtooth_battleship/battleship_cli.py

```diff
@@ -394,44 +394,70 @@ def do_show(args, config):
     else:
         raise BattleshipException("Player hasn't joined game.")
 
+    # figure out who fired last and who is calling do_show
+    # to determine which board * is diplayed on to
+    # show pending shot
+    if 'LastFireRow' in game and 'LastFireColumn' in game:
+        last_fire = (int(game['LastFireRow']) - 1, int(ord(game['LastFireColumn']))-ord('A'))
+    else:
+        last_fire = None
+
+    if game_state == 'P1-NEXT' and target_board_name == 'TargetBoard1':
+        # player 2 last shot and player 1 is looking
+        will_be_on_target_board = False
+    elif game_state == 'P1-NEXT' and target_board_name == 'TargetBoard2':
+        # player 2 last shot and player 2 is looking
+        will_be_on_target_board = True
+    elif game_state == 'P2-NEXT' and target_board_name == 'TargetBoard1':
+        # player 1 last shot and player 1 is looking
+        will_be_on_target_board = True
+    elif game_state == 'P2-NEXT' and target_board_name == 'TargetBoard2':
+        # player 1 last shot and player 2 is looking
+        will_be_on_target_board = False
+    else:
+        last_fire = None
+        will_be_on_target_board = False
+
+
     if target_board_name in game:
         target_board = game[target_board_name]
         size = len(target_board)
+        if will_be_on_target_board and last_fire:
+            target_board[last_fire[0]][last_fire[1]] = '* '
```

Comment:
For clarity, suggest we treat target_board[] as read-only and render the '* ' within the print_board function.